cmake/modules/FindIconv.cmake

```diff
@@ -31,14 +31,11 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
     # Libc causes grief for linux, so search if found library is libc.* and only
     # create imported TARGET if its not
     if(NOT ${ICONV_LIBRARY} MATCHES ".*libc\..*")
-      add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} UNKNOWN IMPORTED)
-      set_target_properties(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} PROPERTIES
-                                                                       IMPORTED_LOCATION "${ICONV_LIBRARY}"
-                                                                       INTERFACE_INCLUDE_DIRECTORIES "${ICONV_INCLUDE_DIR}")
-    endif()
-  else()
-    if(Iconv_FIND_REQUIRED)
-      message(FATAL_ERROR "Iconv libraries were not found.")
+      add_library(Iconv::Iconv UNKNOWN IMPORTED)
+      set_target_properties(Iconv::Iconv PROPERTIES
+                                         IMPORTED_LOCATION "${ICONV_LIBRARY}"
+                                         INTERFACE_INCLUDE_DIRECTORIES "${ICONV_INCLUDE_DIR}")
+      set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP Iconv::Iconv)
```

Comment:
Don't go changing this back. This needs another solution

Reply:
Can you please let me know how to handle it? I just cherry picked your change from months ago. Cmake is black magic to me

Reply:
Yeah will do when I get some time to think about it. Will try to not take too long

Reply:
https://github.com/fuzzard/xbmc/commit/27924f4082b0a64e2d75766c7cddd95c7df8930d for the standardisation of the FindExiv.cmake file

https://github.com/fuzzard/xbmc/commit/3fd59dff8690cff660e29adcbe24e76179fd8c28 for the iconv requirements. just put it as one of the first commits in the PR. Alternatively i can PR this separately, but the approval turn around will probably be slow causing this PR to be held up.